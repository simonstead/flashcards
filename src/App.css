@import url('https://fonts.googleapis.com/css?family=Montserrat');

.App {
  text-align: center;
  font-family: 'Montserrat', sans-serif;
}

.Title {
  color: white;
  border-bottom: 1px solid white;
  width: 10em;
  margin: 1em auto;
}

.Flashcard {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 1em;
  height: 60vh;
}

.English,
.Spanish {
  border: 1px solid #efefef;
  width: 15em;
  height: 15em;
  padding: 1em;
  margin: 1em;
  border-radius: 1em;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  box-shadow: 1px 1px 1px 0px;
  background-color: white;
}

.Rank {
  flex-grow: 1;
}

.Word {
  flex-grow: 2;
}

.Button {
  box-shadow: 1px 1px 1px 0px;
  padding: 1em;
  width: 6em;
  border-radius: 0.5em;
  margin: 2em auto;
  cursor: pointer;
  background-color: white;
}

@media only screen and (min-width: 600px) {
  .Flashcard {
    flex-direction: row;
  }
}
