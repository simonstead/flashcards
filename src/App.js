import React, { Component } from 'react';
import './App.css';
import Data from './data/spanish_words.json';
import { randomWord } from './helpers';

const Flashcard = ({ word }) => (
  <div className="Flashcard">
    <div className="Spanish">
      <div className="Rank">{word.Rank}</div>
      <h1 className="Word">{word.Spanish}</h1>
    </div>
    <div className="English">
      <div className="Rank">{word.Rank}</div>
      <h1 className="Word">{word.English}</h1>
    </div>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      word: randomWord(Data)
    };
  }

  render() {
    return (
      <div className="App">
        <h1 className="Title">Spanish flashcards</h1>
        <Flashcard word={this.state.word} />
        <div className="Button">
          <a onClick={() => this.setState({ word: randomWord(Data) })}>Next</a>
        </div>
      </div>
    );
  }
}

export default App;
