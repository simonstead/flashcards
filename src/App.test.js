import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Data from './data/spanish_words.json';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('can read data', () => {
  expect(Data[0]).toMatchSnapshot();
});

it('can get a random word', () => {
  const randomWord = words => words[Math.floor(Math.random() * words.length)];
  expect(parseInt(randomWord(Data).Rank, 10)).toBeGreaterThan(0);
});
