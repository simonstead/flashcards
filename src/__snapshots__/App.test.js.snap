// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can read data 1`] = `
Object {
  "English": "from",
  "Rank": "1",
  "Spanish": "de",
}
`;
